src/main/java/spoon/support/compiler/jdt/ParentExiter.java

```diff
@@ -973,8 +976,33 @@ private CtCatch getLastCatcher(CtTry tryBlock) {
 
 	@Override
 	public void visitCtTryWithResource(CtTryWithResource tryWithResource) {
-		if (child instanceof CtResource) {
-			tryWithResource.addResource((CtResource<?>) child);
+		if (child instanceof CtLocalVariable) {
+			// normal, happy path of declaring a new variable
+			tryWithResource.addResource((CtLocalVariable) child);
+		} else if (child instanceof CtVariableRead) {
+			// special case of the resource being declared before
+			final CtVariableReference<?> variableRef = ((CtVariableRead<?>) child).getVariable();
```

Comment:
Are there any reasons for the final modifiers, especially for `Collections`?

Reply:
the IDE puts it automatically, and it does not hurt.